### ISSUE TITLE
Wiki

### ISSUE THREAD

**Issue by maruf-bepary-commerzbank** [mbeps/llama3.1_fine-tuning_mult-it#3]
# LLaMA 3.1 8B — Italian Fine-Tuning Overview

## Overview

This project fine-tunes an Italian-adapted LLaMA 3.1 8B checkpoint on native Italian multiple-choice question-answering data, with the goal of improving the model's accuracy on the ITALIC benchmark — a 10,000-question Italian cultural and linguistic evaluation suite. It forms **Iteration 1** of a three-iteration experimental study (*Exploring Modern Reasoning-Enabled Large Language Models for Language Understanding*, King's College London, MSc Artificial Intelligence, 7CCSMPRJ), establishing the **non-reasoning SFT baseline** against which reasoning-capable models are subsequently compared.

The central research question is whether standard supervised fine-tuning — without any chain-of-thought scaffolding or reasoning augmentation — can meaningfully close the gap between a general-purpose multilingual model and a model that genuinely understands Italian language and culture. This pipeline is deliberately the simplest in the project: no thinking tokens, no hybrid training modes, no reasoning toggle. It provides the control condition that makes later comparisons meaningful.

---

## The Base Model

The starting point is **LLaMA 3.1 8B ITA** (`DeepMount00/Llama-3.1-8b-ITA`), an Italian-adapted derivative of Meta's LLaMA 3.1 8B. Meta's LLaMA 3.1 was the first LLaMA generation to formally support multiple languages beyond English, with Italian among the eight officially supported languages. The 8B variant is a dense decoder-only transformer — meaning it processes tokens sequentially and predicts the next token at each step — built around several modern architectural refinements: Grouped Query Attention (which reduces memory usage during inference), Rotary Position Embeddings (which encode relative position between tokens), SwiGLU feed-forward activations, and RMSNorm for numerical stability.

The Italian-adapted checkpoint was chosen rather than the standard multilingual base because it already carries stronger Italian linguistic priors from prior adaptation work, providing a more meaningful starting line for cultural and linguistic fine-tuning. At baseline, it achieves approximately 70% accuracy on ITALIC — already ahead of the multilingual base, but with clear headroom, particularly on language capability tasks such as morphology, orthography, and syntax.

---

## The Training Dataset

The fine-tuning corpus is **Mult-IT** (*Multiple Choice Questions on Multiple Topics in Italian*), introduced by Rinaldi et al. as part of the CALAMITA shared task at EVALITA 2023 and formally described at CLiC-it 2024. Mult-IT is a large-scale collection of approximately 100,000 multiple-choice questions drawn from authentic Italian public competitive examinations — university entrance materials, public-sector employment exams, and professional qualification tests — written natively in Italian rather than translated from English.

This native provenance is significant. Most Italian language resources in large LLM training corpora are either sparse subsets of English-dominated web crawls or machine-translated from English originals, both of which introduce Anglo-centric bias and unnatural phrasing. Mult-IT avoids these artefacts entirely. Its domain coverage across twelve subject areas — spanning humanities, law, science, medicine, civic education, and language proficiency — maps closely onto the ITALIC benchmark's own evaluation schema, making it a well-matched training signal. The dataset was used under a research-only licence with explicit permission from the authors, and is not publicly redistributed.

---

## Fine-Tuning Approach

The fine-tuning technique is **LoRA (Low-Rank Adaptation)**, applied without quantisation. LoRA works by freezing all of the original model's weights and inserting small, trainable matrices — called adapters — into the model's linear projection layers. During training, only these adapter matrices are updated; the vast majority of the model's parameters remain completely unchanged. Once training is complete, the adapters can be mathematically merged back into the base weights, leaving a single model with no inference overhead.

The decision not to quantise (i.e., not to use QLoRA) preserves full numerical precision throughout training, avoiding the subtle accuracy degradation that 4-bit quantisation can introduce — a reasonable trade-off given the availability of sufficient GPU memory. Adapters are inserted across all seven linear projection layers in each transformer block, covering both the attention mechanism and the feed-forward network, giving the adaptation broad coverage of the model's representational capacity.

```mermaid
flowchart TD
    A[Native Italian MCQA\nMult-IT Dataset] --> B[Prompt Formatting\nItalian MCQ Template]
    B --> C[Base Model\nLLaMA 3.1 8B ITA\nfrozen weights]
    C --> D[LoRA Adapters\ninjected into 7 layers]
    D --> E[SFT Training\n1 epoch · BF16 precision]
    E --> F[Trained Adapter Weights\nsaved separately]
    F --> G[Merge Adapters\ninto base model]
    G --> H[Fine-Tuned Model\nready for evaluation]
```

---

## Training Process

Training follows a single-stage supervised fine-tuning workflow. The dataset is loaded, analysed for category and answer-label distribution, and formatted into a plain Italian prompt structure — each question presented in Italian, followed by labelled answer choices, with the correct letter appended as the target. No system prompt, chat template, or special role tokens are used; this is a deliberate design choice that reflects the simplicity of the baseline.

The formatted examples are packed end-to-end into fixed-length token sequences to maximise GPU utilisation, then fed to the trainer for a single training epoch. A cosine learning rate schedule with a short warm-up period prevents large gradient updates from destabilising the randomly initialised adapter matrices early in training. The model learns, in effect, to associate the structure of Italian exam questions with their correct answers — and in doing so, internalises patterns of Italian grammar, vocabulary, and cultural knowledge that improve generalisation to unseen ITALIC questions.

---

## Results

Fine-tuning on Mult-IT produces a measurable improvement across both of ITALIC's macro-categories:

| Metric                | Before Fine-Tuning | After Fine-Tuning | Change      |
| --------------------- | ------------------ | ----------------- | ----------- |
| ITALIC Overall        | ~70.3%             | ~73.9%            | **+3.6 pp** |
| Language Capability   | ~65.5%             | ~70.3%            | **+4.8 pp** |
| Culture & Commonsense | ~73.1%             | ~75.8%            | **+2.7 pp** |

The gain is largest on **Language Capability** tasks — morphology, orthography, syntax, lexicon — which is consistent with the hypothesis that smaller models have the most headroom on Italian linguistic structure, where the pre-training data imbalance relative to English is most acute. Culture and Commonsense tasks also improve, reflecting the broad domain coverage of the Mult-IT training corpus. The fine-tuned model produces a single answer letter directly, with no intermediate reasoning steps.

---

## Connection to the Wider Project

This pipeline is the simplest in the study: pure supervised fine-tuning on a single dataset, no chain-of-thought, no reasoning mode. It establishes the **SFT ceiling for a non-reasoning model** — the best accuracy achievable through standard fine-tuning alone — before more complex approaches are layered on top in subsequent iterations.

The companion **LLaMA 3.1 8B benchmarking project** evaluates both the base Italian checkpoint and this fine-tuned model on the full ITALIC suite, providing the quantitative comparison that anchors Iteration 1. Results from this baseline feed directly into the cross-model comparisons with **Qwen3 8B** (which introduces hybrid thinking/non-thinking modes in Iteration 2) and **Magistral-Small** (which applies a reasoning-capable architecture in Iteration 3). Together, the three iterations form a controlled study of whether reasoning augmentation — chain-of-thought, thinking tokens, or dedicated reasoning training — adds value over standard SFT for Italian cultural and linguistic understanding.

For implementation detail — dataset engineering, LoRA configuration, training hyperparameters, inference extraction, and model saving — see the [detailed fine-tuning wiki](../llama31-fine-tuning.md).

---

## References

1. Hu, E. J., Shen, Y., Wallis, P., Allen-Zhu, Z., Li, Y., Wang, S., … & Chen, W. (2021). **LoRA: Low-Rank Adaptation of Large Language Models**. *arXiv preprint arXiv:2106.09685*. https://arxiv.org/abs/2106.09685

2. Dettmers, T., Pagnoni, A., Holtzman, A., & Zettlemoyer, L. (2023). **QLoRA: Efficient Finetuning of Quantized LLMs**. *Advances in Neural Information Processing Systems (NeurIPS 2023)*. arXiv:2305.14314. https://arxiv.org/abs/2305.14314

3. Rinaldi, M., Gili, J., Francis, M., Goffetti, M., Patti, V., & Nissim, M. (2024). **Mult-IT: Multiple Choice Questions on Multiple Topics in Italian — A CALAMITA Challenge**. *Proceedings of the Tenth Italian Conference on Computational Linguistics (CLiC-it 2024)*, pp. 1184–1201. ACL Anthology: 2024.clicit-1.131. https://aclanthology.org/2024.clicit-1.131/

4. Seveso, A., Potertì, D., Federici, E., Mezzanzanica, M., & Mercorio, F. (2025). **ITALIC: An Italian Culture-Aware Natural Language Benchmark**. *Proceedings of NAACL 2025 (Long Papers)*, pp. 1469–1478. DOI: 10.18653/v1/2025.naacl-long.68. https://aclanthology.org/2025.naacl-long.68.pdf

5. DeepMount00. (2024). **Llama-3.1-8b-ITA** [Model]. Hugging Face Hub. https://huggingface.co/DeepMount00/Llama-3.1-8b-ITA

6. Meta AI. (2024). **The Llama 3 Herd of Models**. *arXiv preprint arXiv:2407.21783*. https://arxiv.org/abs/2407.21783

7. Loshchilov, I., & Hutter, F. (2016). **SGDR: Stochastic Gradient Descent with Warm Restarts**. *arXiv preprint arXiv:1608.03983*. https://arxiv.org/abs/1608.03983

8. Micikevicius, P., Narang, S., Alben, J., Diamos, G., Elsen, E., Garcia, D., … & Wu, H. (2017). **Mixed Precision Training**. *arXiv preprint arXiv:1710.03740*. https://arxiv.org/abs/1710.03740
